draft-irtf-cfrg-hash-to-curve.md

```diff
@@ -533,44 +738,53 @@ document are to be interpreted as described in {{RFC2119}}.
-for an integer m>1.
-
-Elliptic curves can be represented by equations of different standard forms,
-including, but not limited to Weierstrass, Montgomery, and Edwards. Each
-of these variants correspond to a different category of curve equation.
-Certain encoding functions may have requirements on the curve form, the
-characteristic of the field, and the parameters of the curve.
-
-An elliptic curve E is specified by an equation and a finite field F. The curve
-E forms a group, whose elements are points who satisfy the curve equation,
-and the coordinates of a point are elements of F. As a group, E has order n,
-which is the number of points on the curve. For security reasons, it is a
-strong requirement that all cryptographic operations take place in a prime
-order group. However, not all elliptic curves generate groups of prime order.
-For example, some elliptic curves have order n = h\*r, where r is a large prime,
-and h is a non-negative integer known as the cofactor. Thus, one may wish an
-encoding that returns points on a subgroup of order r. To this end, the cofactor
-clearing process consists on multiplying a point P on E by the cofactor h, this guarantees that hP is a point belonging to a subgroup of order r.
+## Elliptic curves {#bg-curves}
+
+The following is a brief definition of elliptic curves, with an emphasis on
+important parameters and their relation to hashing to curves.
+
+Let F be the finite field GF(q) of prime characteristic p. In most cases F
+is a prime field, so q=p. Otherwise, F is a field extension, so q=p^m for
+an integer m > 1. This document assumes that elements of field extensions
+are written in a primitive element or polynomial basis, i.e., as vectors
+of m elements of GF(p) where vector elements are written in ascending order
+by degree. For example, if q=p^2 and the primitive element basis is {1, i},
+then the vector (a, b) corresponds to the element a + b * i.
+
+An elliptic curve E is specified by a cubic equation in two variables and a
+finite field F. An elliptic curve equation takes one of several standard forms,
+including (but not limited to) Weierstrass, Montgomery, and Edwards. Each of
+these forms defines a category of curve equations that is sometimes called a
+"curve shape."
+
+The curve E forms an algebraic group whose elements are the points (x, y)
+satisfying the curve equation, where x and y are elements of F. This group
+has order n, meaning that there are n distinct points (x, y). In general,
+security of cryptographic primitives requires using a group of prime order.
+However, not all elliptic curves induce groups of prime order: most elliptic
+curves have order n = h * r, where r is a large prime and h is a non-negative
+integer called the cofactor. Thus, for cryptographic applications a hash
+function to the curve E should return points in the subgroup of order r. For
+a point not in the prime-order subgroup, the process of mapping to a point
```

Comment:
Sorry, I don't quite understand the issue here. Can you please explain?

Multiplying any point by h will return a point in the subgroup of order r. Of course, it might return the identity point, but that's fine---it's also in the subgroup. So it's not obvious to me why it's necessary to talk about points having order a multiple of r.

Sorry if I'm not understanding...

Reply:
In this line of the document it seems that cofactor clearing only works for points that are not in the prime-order subgroup. 
I was trying to be more general saying that cofactor clearing works for any point in the curve.